openmdao/components/tests/test_meta_model_unstructured_comp.py

```diff
@@ -1244,5 +1245,178 @@ def test_warning_bug(self):
+
+    def test_nearest_neighbor(self):
+        import numpy as np
+
+        from openmdao.api import Problem, MetaModelUnStructuredComp, IndepVarComp
+        from openmdao.api import NearestNeighbor
+
+        prob = Problem()
+
+        prob.model.add_subsystem('p', IndepVarComp('x', 2.1))
+
+        sin_mm = MetaModelUnStructuredComp()
+        sin_mm.add_input('x', 0.)
+        sin_mm.add_output('f_x', 0., surrogate=NearestNeighbor(interpolant_type='linear'))
+
+        prob.model.add_subsystem('sin_mm', sin_mm)
+
+        prob.model.connect('p.x', 'sin_mm.x')
+
+        prob.setup(check=True)
+
+        # train the surrogate and check predicted value
+        sin_mm.options['train:x'] = np.linspace(0,10,20)
+        sin_mm.options['train:f_x'] = .5*np.sin(sin_mm.options['train:x'])
+
+        prob['sin_mm.x'] = 2.1
+
+        prob.run_model()
+
+        assert_rel_error(self, prob['sin_mm.f_x'], .5*np.sin(prob['sin_mm.x']), 2e-3)
+
+    def test_response_surface(self):
+        import numpy as np
+
+        from openmdao.api import Problem, MetaModelUnStructuredComp, IndepVarComp
+        from openmdao.api import ResponseSurface
+
+        prob = Problem()
+
+        prob.model.add_subsystem('p', IndepVarComp('x', 2.1))
+
+        sin_mm = MetaModelUnStructuredComp()
+        sin_mm.add_input('x', 0.)
+        sin_mm.add_output('f_x', 0., surrogate=ResponseSurface())
+
+        prob.model.add_subsystem('sin_mm', sin_mm)
+
+        prob.model.connect('p.x', 'sin_mm.x')
+
+        prob.setup(check=True)
+
+        # train the surrogate and check predicted value
+        sin_mm.options['train:x'] = np.linspace(0, 3.14, 20)
+        sin_mm.options['train:f_x'] = .5*np.sin(sin_mm.options['train:x'])
+
+        prob['sin_mm.x'] = 2.1
+
+        prob.run_model()
+
+        assert_rel_error(self, prob['sin_mm.f_x'], .5*np.sin(prob['sin_mm.x']), 2e-3)
+
+    def test_kriging_options_eval_rmse(self):
+        import numpy as np
+
+        from openmdao.api import Problem, MetaModelUnStructuredComp, IndepVarComp
+        from openmdao.api import KrigingSurrogate
+
+        prob = Problem()
+
+        prob.model.add_subsystem('p', IndepVarComp('x', 2.1))
+
+        sin_mm = MetaModelUnStructuredComp()
+        sin_mm.add_input('x', 0.)
+        sin_mm.add_output('f_x', 0., surrogate=KrigingSurrogate(eval_rmse=True))
+
+        prob.model.add_subsystem('sin_mm', sin_mm)
+
+        prob.model.connect('p.x', 'sin_mm.x')
+
+        prob.setup(check=True)
+
+        # train the surrogate and check predicted value
+        sin_mm.options['train:x'] = np.linspace(0,10,20)
+        sin_mm.options['train:f_x'] = .5*np.sin(sin_mm.options['train:x'])
+
+        prob['sin_mm.x'] = 2.1
+
+        prob.run_model()
+
+        print("mean")
+        assert_rel_error(self, prob['sin_mm.f_x'], .5*np.sin(prob['sin_mm.x']), 1e-4)
+        print("std")
+        assert_rel_error(self, sin_mm._metadata('f_x')['rmse'][0, 0], 0.0, 1e-4)
```

Comment:
This is telling the user to use a private (underscored) function to access the metadata...  should this be elevated to be part of the API?

Reply:
We need to modify MetamodelUnstructured to provide the rmse as an openmdao output, but that is going to take some thought and re-design.